src/maplibregl-controller.ts

```diff
@@ -81,56 +85,68 @@ export function createMapLibreGlMapController(
 
   let savedData: GeoJSON | undefined; // used to restore features on style switch
 
-  function addFullGeometryLayer() {
+  function syncFullGeometryLayer() {
+    if (!map.loaded) {
+      map.once("load", syncFullGeometryLayer);
+
+      return;
+    }
+
     const effFullGeometryStyle = !fullGeometryStyle
       ? undefined
       : fullGeometryStyle === true
         ? defaultGeometryStyle
         : fullGeometryStyle;
 
-    if (
-      !map.getSource("full-geom") &&
-      (effFullGeometryStyle?.fill || effFullGeometryStyle?.line)
-    ) {
-      map.addSource("full-geom", {
+    if (!effFullGeometryStyle?.fill && !effFullGeometryStyle?.line) {
+      return;
+    }
+
+    const source = map.getSource(RESULT_SOURCE) as GeoJSONSource;
```

Comment:
doesn't really make any difference but `getSource` is generic so `const source = map.getSource<GeoJSONSource>(RESULT_SOURCE);` seems to work

Reply:
Updated, thanks.